roles/users/tasks/main.yml

```diff
@@ -50,7 +50,7 @@
   changed_when: false
   sudo: no
   register: admin_user_status
-  when: not sshd_permit_root_login
+  when: not sshd_permit_root_login and ansible_ssh_host is defined
```

Comment:
The results you are seeing seem like what is expected. My results seem unexpected (Ansible 1.9.2 & 1.9.3). So, I'd trust yours. I'm guessing my problem is that I'm not testing via Packer, but on my local mac. Or I simply may not be understanding how things work. 

I create an entry in my `hosts/staging` file that shouldn't be able to connect via SSH because there's no IP to help the connection succeed. So, `ansible_connection` should remain `local`. It shouldn't be set back to undefined due to a successful SSH connection.
 ```
[web]
fake
```
Then I run `ansible-playbook server.yml -i hosts/staging --connection=local`. I've made the first couple tasks check the value of the variable: 
```
- debug: var=ansible_connection
- local_action: debug msg="{{ ansible_connection }}"
```
They both come out undefined. I'm probably missing something.